src/btree/bt_vrfy.c

```diff
@@ -268,6 +275,17 @@ __wt_verify(WT_SESSION_IMPL *session, const char *cfg[])
             WT_WITH_PAGE_INDEX(
               session, ret = __verify_tree(session, &btree->root, &addr_unpack, vs));
 
+            /*
+             * The checkpoints are in time-order, so the last one in the list is the most recent. If
+             * this is the most recent checkpoint, verify the history store against it.
+             */
+            if (ret == 0 && (ckpt + 1)->name == NULL && !skip_hs) {
+                /* Open a history store cursor. */
+                WT_ERR(__wt_hs_cursor(session, &session_flags, &is_owner));
+                WT_TRET(__wt_history_store_verify_one(session));
+                WT_TRET(__wt_hs_cursor_close(session, session_flags, is_owner));
```

Comment:
I was thinking we needed to do the later calls (exclusive and evicting the file with `WT_SYNC_DISCARD` and unloading the checkpoint) before calling WT_ERR. is it safe to skip all of those?

Reply:
I confirmed, I need to perform the other calls (at least the exclusive call) because if I don't and error out too soon, I trigger an assertion `[1587582440:623658][25469:0x7fbb7d7fa700], t, file:wt, WT_SESSION.verify: __wt_evict_file_exclusive_off, 944: v >= 0`. If I error out a few lines further down, then it panics correctly.

Reply:
I have added a comment explaining this at that location instead.